### PR TITLE
r/aws_lambda_provisioned_concurrency_configuration: add `skip_destroy` argument

### LINKED ISSUE #31645: [Enhancement]: Add ability to skip destruction of Lambda Provisioned Concurrency Configurations
### Description

As currently implemented, provisioned concurrency configurations are always removed when a lambda version is updated (changes to the `qualifier` argument force replacement). To enable preservation of provisioned concurrency configurations on previous versions, a `skip_destroy` argument should be added similar to the [aws_lambda_function](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#skip_destroy) and [aws_lambda_layer_version](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_layer_version#skip_destroy) resources.

### Affected Resource(s) and/or Data Source(s)

- aws_lambda_provisioned_concurrency_configuration

### Potential Terraform Configuration

```terraform
resource "aws_lambda_provisioned_concurrency_config" "test" {
  function_name                     = aws_lambda_function.test.function_name
  provisioned_concurrent_executions = 1
  qualifier                         = aws_lambda_function.test.version

  skip_destroy = true
}
```

### References

_No response_

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by jar-b**
### Description
Adds a `skip_destroy` argument, allowing the preservation of provisioned concurrency configurations when lambda function versions are updated.

### Relations
Closes #31645

### Output from Acceptance Testing

```console
$ SKIP_DESTROY_ENABLED=true make testacc PKG=lambda TESTS=TestAccLambdaProvisionedConcurrencyConfig_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lambda/... -v -count 1 -parallel 20 -run='TestAccLambdaProvisionedConcurrencyConfig_'  -timeout 180m
=== RUN   TestAccLambdaProvisionedConcurrencyConfig_basic
=== PAUSE TestAccLambdaProvisionedConcurrencyConfig_basic
=== RUN   TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaFunction
=== PAUSE TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaFunction
=== RUN   TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaProvisionedConcurrency
=== PAUSE TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaProvisionedConcurrency
=== RUN   TestAccLambdaProvisionedConcurrencyConfig_provisionedConcurrentExecutions
=== PAUSE TestAccLambdaProvisionedConcurrencyConfig_provisionedConcurrentExecutions
=== RUN   TestAccLambdaProvisionedConcurrencyConfig_Qualifier_aliasName
=== PAUSE TestAccLambdaProvisionedConcurrencyConfig_Qualifier_aliasName
=== RUN   TestAccLambdaProvisionedConcurrencyConfig_skipDestroy
=== PAUSE TestAccLambdaProvisionedConcurrencyConfig_skipDestroy
=== CONT  TestAccLambdaProvisionedConcurrencyConfig_basic
=== CONT  TestAccLambdaProvisionedConcurrencyConfig_provisionedConcurrentExecutions
=== CONT  TestAccLambdaProvisionedConcurrencyConfig_skipDestroy
=== CONT  TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaProvisionedConcurrency
=== CONT  TestAccLambdaProvisionedConcurrencyConfig_Qualifier_aliasName
=== CONT  TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaFunction
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_skipDestroy (187.75s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaFunction (201.64s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaProvisionedConcurrency (206.41s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_basic (206.54s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_Qualifier_aliasName (222.91s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_provisionedConcurrentExecutions (373.41s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/lambda     376.666s
```

### CODE REVIEW

File: internal/service/lambda/provisioned_concurrency_config_test.go [hunk 1]
```diff
@@ -159,9 +163,40 @@ func TestAccLambdaProvisionedConcurrencyConfig_Qualifier_aliasName(t *testing.T)
...
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	lambdaFunctionResourceName := "aws_lambda_function.test"
+	resourceName := "aws_lambda_provisioned_concurrency_config.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, names.LambdaEndpointID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckProvisionedConcurrencyConfigNoDestroy(ctx),
+		Steps: []resource.TestStep{
```
Comment (gdavison) [inline internal/service/lambda/provisioned_concurrency_config_test.go:190]: We could test this without the environment variable by using the following test steps, which would also test the originally reported problem

1. Create
2. Publish a new version of the function, but validate that the old version of the function still has concurrency set
3. Delete if from the new version, and validate that it's still set on the new version

**Review by gdavison — COMMENTED** [review]
Looks good. I have a comment about the added test

File: internal/service/lambda/provisioned_concurrency_config_test.go (hunk 1 above)
Comment (jar-b) [reply internal/service/lambda/provisioned_concurrency_config_test.go:189]: Good suggestion - I've updated to this pattern and we shouldn't have any dangling resources leftover now.

**Comment by jar-b** [comment]
Test results post `_skipDestroy` updates:

```console
$ make testacc PKG=lambda TESTS=TestAccLambdaProvisionedConcurrencyConfig_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lambda/... -v -count 1 -parallel 20 -run='TestAccLambdaProvisionedConcurrencyConfig_'  -timeout 180m

--- PASS: TestAccLambdaProvisionedConcurrencyConfig_Qualifier_aliasName (173.96s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaFunction (188.80s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_Disappears_lambdaProvisionedConcurrency (205.29s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_basic (260.23s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_skipDestroy (347.22s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_provisionedConcurrentExecutions (437.56s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/lambda     440.827s
```

**Pull request merged**